System.ObjectDisposedException: Cannot access a disposed object 'Xamarin.Forms.Platform.Android.FastRenderers.LabelRenderer'

_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/content/problem/1035354/systemobjectdisposedexception-cannot-access-a-disp-1.html)._

---
After upgrading to the Xamarin Forms 4.6.0726, starting getting frequent intermittent crash on Android. Upon investigation, found that this is coming from Xamarin's LabelRenderer

05-17 07:27:06.656 E/mono    ( 5133): 
05-17 07:27:06.656 E/mono    ( 5133): Unhandled Exception:
05-17 07:27:06.656 E/mono    ( 5133): System.ObjectDisposedException: Cannot access a disposed object.
05-17 07:27:06.656 E/mono    ( 5133): Object name: 'Xamarin.Forms.Platform.Android.FastRenderers.LabelRenderer'.
05-17 07:27:06.656 E/mono    ( 5133):   at (wrapper dynamic-method) Android.Runtime.DynamicMethodNameCounter.29(intptr,intptr)
05-17 07:27:06.656 E/mono    ( 5133):   at (wrapper native-to-managed) Android.Runtime.DynamicMethodNameCounter.29(intptr,intptr)
05-17 07:27:06.656 E/mono-rt ( 5133): [ERROR] FATAL UNHANDLED EXCEPTION: System.ObjectDisposedException: Cannot access a disposed object.
05-17 07:27:06.656 E/mono-rt ( 5133): Object name: 'Xamarin.Forms.Platform.Android.FastRenderers.LabelRenderer'.
05-17 07:27:06.656 E/mono-rt ( 5133):   at (wrapper dynamic-method) Android.Runtime.DynamicMethodNameCounter.29(intptr,intptr)
05-17 07:27:06.656 E/mono-rt ( 5133):   at (wrapper native-to-managed) Android.Runtime.DynamicMethodNameCounter.29(intptr,intptr)
05-17 07:27:06.698 D/        ( 5133): HostConnection::get() New Host Connection established 0xbd1b1240, tid 5133

Not sure where it is coming from. I have ListView as main view and wondering something is wrong with  newer version of ListView causing leak and crash.

I don't have a custom label renderer so it has to be coming from Xamarin's platform renderer
Please suggest.

---
### Original Comments

#### Visual Studio Feedback System on 5/17/2020, 11:54 PM: 

<p>We have directed your feedback to the appropriate engineering team for further evaluation. The team will review the feedback and notify you about the next steps.</p>


---
### Original Solutions
(no solutions)

I'm facing same error in my project. I have Xamarin.Forms 4.6.0.726 version. 
Stack trace:
```
  at Java.Interop.JniPeerMembers.AssertSelf (Java.Interop.IJavaPeerable self) [0x00029] in <dac4c5a4b77f4e61a5e6d9d3050dfb9f>:0 
  at Java.Interop.JniPeerMembers+JniInstanceMethods.InvokeNonvirtualObjectMethod (System.String encodedMember, Java.Interop.IJavaPeerable self, Java.Interop.JniArgumentValue* parameters) [0x00000] in <dac4c5a4b77f4e61a5e6d9d3050dfb9f>:0 
  at Android.Views.View.get_Context () [0x00000] in /Users/builder/azdo/_work/204/s/xamarin-android/src/Mono.Android/obj/Release/monoandroid10/android-29/mcw/Android.Views.View.cs:5598 
  at Xamarin.Forms.Platform.Android.Platform.GetNativeSize (Xamarin.Forms.VisualElement view, System.Double widthConstraint, System.Double heightConstraint) [0x00019] in D:\a\1\s\Xamarin.Forms.Platform.Android\Platform.cs:1341 
  at Xamarin.Forms.Forms+AndroidPlatformServices.GetNativeSize (Xamarin.Forms.VisualElement view, System.Double widthConstraint, System.Double heightConstraint) [0x00000] in D:\a\1\s\Xamarin.Forms.Platform.Android\Forms.cs:930 
  at Xamarin.Forms.VisualElement.OnSizeRequest (System.Double widthConstraint, System.Double heightConstraint) [0x00025] in D:\a\1\s\Xamarin.Forms.Core\VisualElement.cs:800 
  at Xamarin.Forms.VisualElement.OnMeasure (System.Double widthConstraint, System.Double heightConstraint) [0x00000] in D:\a\1\s\Xamarin.Forms.Core\VisualElement.cs:783 
  at Xamarin.Forms.VisualElement.GetSizeRequest (System.Double widthConstraint, System.Double heightConstraint) [0x00053] in D:\a\1\s\Xamarin.Forms.Core\VisualElement.cs:659 
  at Xamarin.Forms.VisualElement.Measure (System.Double widthConstraint, System.Double heightConstraint, Xamarin.Forms.MeasureFlags flags) [0x00054] in D:\a\1\s\Xamarin.Forms.Core\VisualElement.cs:717 
  at Xamarin.Forms.Grid.CalculateAutoCells (System.Double width, System.Double height) [0x000e5] in D:\a\1\s\Xamarin.Forms.Core\GridCalc.cs:133 
  at Xamarin.Forms.Grid.MeasureGrid (System.Double width, System.Double height, System.Boolean requestSize) [0x0000c] in D:\a\1\s\Xamarin.Forms.Core\GridCalc.cs:485 
  at Xamarin.Forms.Grid.OnSizeRequest (System.Double widthConstraint, System.Double heightConstraint) [0x0002a] in D:\a\1\s\Xamarin.Forms.Core\GridCalc.cs:60 
  at Xamarin.Forms.VisualElement.OnMeasure (System.Double widthConstraint, System.Double heightConstraint) [0x00000] in D:\a\1\s\Xamarin.Forms.Core\VisualElement.cs:783 
  at Xamarin.Forms.VisualElement.GetSizeRequest (System.Double widthConstraint, System.Double heightConstraint) [0x00053] in D:\a\1\s\Xamarin.Forms.Core\VisualElement.cs:659 
  at Xamarin.Forms.Layout.GetSizeRequest (System.Double widthConstraint, System.Double heightConstraint) [0x00000] in D:\a\1\s\Xamarin.Forms.Core\Layout.cs:132 
  at Xamarin.Forms.VisualElement.Measure (System.Double widthConstraint, System.Double heightConstraint, Xamarin.Forms.MeasureFlags flags) [0x00054] in D:\a\1\s\Xamarin.Forms.Core\VisualElement.cs:717 
  at Xamarin.Forms.TemplatedView.OnSizeRequest (System.Double widthConstraint, System.Double heightConstraint) [0x0003c] in D:\a\1\s\Xamarin.Forms.Core\TemplatedView.cs:43 
  at Xamarin.Forms.VisualElement.OnMeasure (System.Double widthConstraint, System.Double heightConstraint) [0x00000] in D:\a\1\s\Xamarin.Forms.Core\VisualElement.cs:783 
  at Xamarin.Forms.VisualElement.GetSizeRequest (System.Double widthConstraint, System.Double heightConstraint) [0x00053] in D:\a\1\s\Xamarin.Forms.Core\VisualElement.cs:659 
  at Xamarin.Forms.Layout.GetSizeRequest (System.Double widthConstraint, System.Double heightConstraint) [0x00000] in D:\a\1\s\Xamarin.Forms.Core\Layout.cs:132 
  at Xamarin.Forms.VisualElement.Measure (System.Double widthConstraint, System.Double heightConstraint, Xamarin.Forms.MeasureFlags flags) [0x00054] in D:\a\1\s\Xamarin.Forms.Core\VisualElement.cs:717 
  at Xamarin.Forms.StackLayout.CalculateNaiveLayout (Xamarin.Forms.StackLayout+LayoutInformation layout, Xamarin.Forms.StackOrientation orientation, System.Double x, System.Double y, System.Double widthConstraint, System.Double heightConstraint) [0x000a8] in D:\a\1\s\Xamarin.Forms.Core\StackLayout.cs:163 
  at Xamarin.Forms.StackLayout.CalculateLayout (Xamarin.Forms.StackLayout+LayoutInformation layout, System.Double x, System.Double y, System.Double widthConstraint, System.Double heightConstraint, System.Boolean processExpanders) [0x00058] in D:\a\1\s\Xamarin.Forms.Core\StackLayout.cs:123 
  at Xamarin.Forms.StackLayout.OnSizeRequest (System.Double widthConstraint, System.Double heightConstraint) [0x00019] in D:\a\1\s\Xamarin.Forms.Core\StackLayout.cs:80 
  at Xamarin.Forms.VisualElement.OnMeasure (System.Double widthConstraint, System.Double heightConstraint) [0x00000] in D:\a\1\s\Xamarin.Forms.Core\VisualElement.cs:783 
  at Xamarin.Forms.VisualElement.GetSizeRequest (System.Double widthConstraint, System.Double heightConstraint) [0x00053] in D:\a\1\s\Xamarin.Forms.Core\VisualElement.cs:659 
  at Xamarin.Forms.Layout.GetSizeRequest (System.Double widthConstraint, System.Double heightConstraint) [0x00000] in D:\a\1\s\Xamarin.Forms.Core\Layout.cs:132 
  at Xamarin.Forms.VisualElement.Measure (System.Double widthConstraint, System.Double heightConstraint, Xamarin.Forms.MeasureFlags flags) [0x00054] in D:\a\1\s\Xamarin.Forms.Core\VisualElement.cs:717 
  at Xamarin.Forms.StackLayout.CalculateNaiveLayout (Xamarin.Forms.StackLayout+LayoutInformation layout, Xamarin.Forms.StackOrientation orientation, System.Double x, System.Double y, System.Double widthConstraint, System.Double heightConstraint) [0x000a8] in D:\a\1\s\Xamarin.Forms.Core\StackLayout.cs:163 
  at Xamarin.Forms.StackLayout.CalculateLayout (Xamarin.Forms.StackLayout+LayoutInformation layout, System.Double x, System.Double y, System.Double widthConstraint, System.Double heightConstraint, System.Boolean processExpanders) [0x00058] in D:\a\1\s\Xamarin.Forms.Core\StackLayout.cs:123 
  at Xamarin.Forms.StackLayout.OnSizeRequest (System.Double widthConstraint, System.Double heightConstraint) [0x00019] in D:\a\1\s\Xamarin.Forms.Core\StackLayout.cs:80 
  at Xamarin.Forms.VisualElement.OnMeasure (System.Double widthConstraint, System.Double heightConstraint) [0x00000] in D:\a\1\s\Xamarin.Forms.Core\VisualElement.cs:783 
  at Xamarin.Forms.VisualElement.GetSizeRequest (System.Double widthConstraint, System.Double heightConstraint) [0x00053] in D:\a\1\s\Xamarin.Forms.Core\VisualElement.cs:659 
  at Xamarin.Forms.Layout.GetSizeRequest (System.Double widthConstraint, System.Double heightConstraint) [0x00000] in D:\a\1\s\Xamarin.Forms.Core\Layout.cs:132 
  at Xamarin.Forms.VisualElement.Measure (System.Double widthConstraint, System.Double heightConstraint, Xamarin.Forms.MeasureFlags flags) [0x00054] in D:\a\1\s\Xamarin.Forms.Core\VisualElement.cs:717 
  at Xamarin.Forms.StackLayout.CalculateNaiveLayout (Xamarin.Forms.StackLayout+LayoutInformation layout, Xamarin.Forms.StackOrientation orientation, System.Double x, System.Double y, System.Double widthConstraint, System.Double heightConstraint) [0x000a8] in D:\a\1\s\Xamarin.Forms.Core\StackLayout.cs:163 
  at Xamarin.Forms.StackLayout.CalculateLayout (Xamarin.Forms.StackLayout+LayoutInformation layout, System.Double x, System.Double y, System.Double widthConstraint, System.Double heightConstraint, System.Boolean processExpanders) [0x00058] in D:\a\1\s\Xamarin.Forms.Core\StackLayout.cs:123 
  at Xamarin.Forms.StackLayout.LayoutChildren (System.Double x, System.Double y, System.Double width, System.Double height) [0x0004e] in D:\a\1\s\Xamarin.Forms.Core\StackLayout.cs:58 
  at Xamarin.Forms.Layout.UpdateChildrenLayout () [0x00158] in D:\a\1\s\Xamarin.Forms.Core\Layout.cs:266 
  at Xamarin.Forms.Layout.OnSizeAllocated (System.Double width, System.Double height) [0x0000f] in D:\a\1\s\Xamarin.Forms.Core\Layout.cs:224 
  at Xamarin.Forms.VisualElement.SizeAllocated (System.Double width, System.Double height) [0x00000] in D:\a\1\s\Xamarin.Forms.Core\VisualElement.cs:805 
  at Xamarin.Forms.Layout.ResolveLayoutChanges () [0x0005c] in D:\a\1\s\Xamarin.Forms.Core\Layout.cs:392 
  at Java.Lang.Thread+RunnableImplementor.Run () [0x00008] in /Users/builder/azdo/_work/204/s/xamarin-android/src/Mono.Android/Java.Lang/Thread.cs:36 
  at Java.Lang.IRunnableInvoker.n_Run (System.IntPtr jnienv, System.IntPtr native__this) [0x00008] in /Users/builder/azdo/_work/204/s/xamarin-android/src/Mono.Android/obj/Release/monoandroid10/android-29/mcw/Java.Lang.IRunnable.cs:81 
  at (wrapper dynamic-method) Android.Runtime.DynamicMethodNameCounter.1(intptr,intptr)
```

**Application Output Window show these output:**
> [mono] Unhandled Exception:
> [mono] System.ObjectDisposedException: Cannot access a disposed object.
> [mono] Object name: 'Xamarin.Forms.Platform.Android.FastRenderers.LabelRenderer'.
> [mono]   at (wrapper dynamic-method) Android.Runtime.DynamicMethodNameCounter.1(intptr,intptr)
> [mono]   at (wrapper native-to-managed) Android.Runtime.DynamicMethodNameCounter.1(intptr,intptr)
> [mono-rt] [ERROR] FATAL UNHANDLED EXCEPTION: System.ObjectDisposedException: Cannot access a disposed object.
> [mono-rt] Object name: 'Xamarin.Forms.Platform.Android.FastRenderers.LabelRenderer'.
> [mono-rt]   at (wrapper dynamic-method) Android.Runtime.DynamicMethodNameCounter.1(intptr,intptr)

@PureWeen 
I attach all the information that I get from the output window:

> 09-13 12:37:28.193 I/MonoDroid(21323): UNHANDLED EXCEPTION:
09-13 12:37:28.202 I/MonoDroid(21323): System.ObjectDisposedException: Cannot access a disposed object.
09-13 12:37:28.202 I/MonoDroid(21323): Object name: 'Xamarin.Forms.Platform.Android.Platform+DefaultRenderer'.
09-13 12:37:28.202 I/MonoDroid(21323):   at Java.Interop.JniPeerMembers.AssertSelf (Java.Interop.IJavaPeerable self) [0x00029] in <42748fcc36b74733af2d9940a8f3cc8e>:0 
09-13 12:37:28.202 I/MonoDroid(21323):   at Java.Interop.JniPeerMembers+JniInstanceMethods.InvokeVirtualVoidMethod (System.String encodedMember, Java.Interop.IJavaPeerable self, Java.Interop.JniArgumentValue* parameters) [0x00000] in <42748fcc36b74733af2d9940a8f3cc8e>:0 
09-13 12:37:28.202 I/MonoDroid(21323):   at Android.Views.View.Invalidate () [0x0000a] in <7d2292394f8c488b97f5bc2a0ac0240d>:0 
09-13 12:37:28.202 I/MonoDroid(21323):   at Xamarin.Forms.Forms+AndroidPlatformServices.Invalidate (Xamarin.Forms.VisualElement visualElement) [0x00013] in D:\agent\1\s\Xamarin.Forms.Platform.Android\Forms.cs:932 
09-13 12:37:28.202 I/MonoDroid(21323):   at Xamarin.Forms.Device.Invalidate (Xamarin.Forms.VisualElement visualElement) [0x00001] in D:\agent\1\s\Xamarin.Forms.Core\Device.cs:291 
09-13 12:37:28.202 I/MonoDroid(21323):   at Xamarin.Forms.Layout.OnChildMeasureInvalidated (Xamarin.Forms.VisualElement child, Xamarin.Forms.Internals.InvalidationTrigger trigger) [0x00163] in D:\agent\1\s\Xamarin.Forms.Core\Layout.cs:352 
09-13 12:37:28.202 I/MonoDroid(21323):   at Xamarin.Forms.Layout.OnChildMeasureInvalidated (System.Object sender, System.EventArgs e) [0x00014] in D:\agent\1\s\Xamarin.Forms.Core\Layout.cs:191 
09-13 12:37:28.202 I/MonoDroid(21323):   at Xamarin.Forms.VisualElement.InvalidateMeasureInternal (Xamarin.Forms.Internals.InvalidationTrigger trigger) [0x0000d] in D:\agent\1\s\Xamarin.Forms.Core\VisualElement.cs:844 
09-13 12:37:28.202 I/MonoDroid(21323):   at Xamarin.Forms.Label.Xamarin.Forms.Internals.IFontElement.OnFontSizeChanged (System.Double oldValue, System.Double newValue) [0x00000] in D:\agent\1\s\Xamarin.Forms.Core\Label.cs:247 
09-13 12:37:28.202 I/MonoDroid(21323):   at Xamarin.Forms.FontElement.OnFontSizeChanged (Xamarin.Forms.BindableObject bindable, System.Object oldValue, System.Object newValue) [0x0008f] in D:\agent\1\s\Xamarin.Forms.Core\FontElement.cs:93 
09-13 12:37:28.202 I/MonoDroid(21323):   at Xamarin.Forms.BindableObject.SetValueActual (Xamarin.Forms.BindableProperty property, Xamarin.Forms.BindableObject+BindablePropertyContext context, System.Object value, System.Boolean currentlyApplying, Xamarin.Forms.Internals.SetValueFlags attributes, System.Boolean silent) [0x00165] in D:\agent\1\s\Xamarin.Forms.Core\BindableObject.cs:463 
09-13 12:37:28.202 I/MonoDroid(21323):   at Xamarin.Forms.BindableObject.SetValueCore (Xamarin.Forms.BindableProperty property, System.Object value, Xamarin.Forms.Internals.SetValueFlags attributes, Xamarin.Forms.BindableObject+SetValuePrivateFlags privateAttributes) [0x001cd] in D:\agent\1\s\Xamarin.Forms.Core\BindableObject.cs:397 
09-13 12:37:28.202 I/MonoDroid(21323):   at Xamarin.Forms.BindingExpression.ApplyCore (System.Object sourceObject, Xamarin.Forms.BindableObject target, Xamarin.Forms.BindableProperty property, System.Boolean fromTarget) [0x0028a] in D:\agent\1\s\Xamarin.Forms.Core\BindingExpression.cs:155 
09-13 12:37:28.203 I/MonoDroid(21323):   at Xamarin.Forms.BindingExpression.Apply (System.Boolean fromTarget) [0x0005a] in D:\agent\1\s\Xamarin.Forms.Core\BindingExpression.cs:53 
09-13 12:37:28.203 I/MonoDroid(21323):   at Xamarin.Forms.BindingExpression+BindingExpressionPart.<PropertyChanged>b__49_0 () [0x00000] in D:\agent\1\s\Xamarin.Forms.Core\BindingExpression.cs:742 
09-13 12:37:28.203 I/MonoDroid(21323):   at Xamarin.Forms.BindingExpression+BindingExpressionPart.PropertyChanged (System.Object sender, System.ComponentModel.PropertyChangedEventArgs args) [0x0010f] in D:\agent\1\s\Xamarin.Forms.Core\BindingExpression.cs:753 
09-13 12:37:28.203 I/MonoDroid(21323):   at Xamarin.Forms.BindingExpression+WeakPropertyChangedProxy.OnPropertyChanged (System.Object sender, System.ComponentModel.PropertyChangedEventArgs e) [0x0001b] in D:\agent\1\s\Xamarin.Forms.Core\BindingExpression.cs:646 
09-13 12:37:28.203 I/MonoDroid(21323):   at (wrapper delegate-invoke) <Module>.invoke_void_object_PropertyChangedEventArgs(object,System.ComponentModel.PropertyChangedEventArgs)
09-13 12:37:28.203 I/MonoDroid(21323):   at LabelRender.ViewModels.Base.BaseViewModel.RaiseOnPropertyChanged (System.String propertyName) [0x00001] in D:\Sergio\Programacion\Prototipos\LabelRender\LabelRender\LabelRender\LabelRender\ViewModels\Base\BaseViewModel.cs:26 
09-13 12:37:28.203 I/MonoDroid(21323):   at LabelRender.ViewModels.Base.BaseViewModel.set_SmallFontSize (System.Double value) [0x00008] in D:\Sergio\Programacion\Prototipos\LabelRender\LabelRender\LabelRender\LabelRender\ViewModels\Base\BaseViewModel.cs:16 
09-13 12:37:28.203 I/MonoDroid(21323):   at LabelRender.ContentPageBase.BaseViewPage_SizeChanged (System.Object sender, System.EventArgs e) [0x00057] in D:\Sergio\Programacion\Prototipos\LabelRender\LabelRender\LabelRender\LabelRender\ContentPageBase.xaml.cs:25 
09-13 12:37:28.203 I/MonoDroid(21323):   at Xamarin.Forms.VisualElement.SetSize (System.Double width, System.Double height) [0x00035] in D:\agent\1\s\Xamarin.Forms.Core\VisualElement.cs:1050 
09-13 12:37:28.203 I/MonoDroid(21323):   at Xamarin.Forms.VisualElement.set_Bounds (Xamarin.Forms.Rectangle value) [0x00069] in D:\agent\1\s\Xamarin.Forms.Core\VisualElement.cs:357 
09-13 12:37:28.203 I/MonoDroid(21323):   at Xamarin.Forms.VisualElement.Layout (Xamarin.Forms.Rectangle bounds) [0x00000] in D:\agent\1\s\Xamarin.Forms.Core\VisualElement.cs:727 
09-13 12:37:28.203 I/MonoDroid(21323):   at Xamarin.Forms.Layout.LayoutChildIntoBoundingRegion (Xamarin.Forms.VisualElement child, Xamarin.Forms.Rectangle region) [0x00072] in D:\agent\1\s\Xamarin.Forms.Core\Layout.cs:126 
09-13 12:37:28.203 I/MonoDroid(21323):   at Xamarin.Forms.Page.LayoutChildren (System.Double x, System.Double y, System.Double width, System.Double height) [0x00136] in D:\agent\1\s\Xamarin.Forms.Core\Page.cs:278 
09-13 12:37:28.203 I/MonoDroid(21323):   at Xamarin.Forms.Page.UpdateChildrenLayout () [0x000e6] in D:\agent\1\s\Xamarin.Forms.Core\Page.cs:357 
09-13 12:37:28.203 I/MonoDroid(21323):   at Xamarin.Forms.Page.OnSizeAllocated (System.Double width, System.Double height) [0x00011] in D:\agent\1\s\Xamarin.Forms.Core\Page.cs:337 
09-13 12:37:28.203 I/MonoDroid(21323):   at Xamarin.Forms.VisualElement.SizeAllocated (System.Double width, System.Double height) [0x00000] in D:\agent\1\s\Xamarin.Forms.Core\VisualElement.cs:817 
09-13 12:37:28.203 I/MonoDroid(21323):   at Xamarin.Forms.Page.ForceLayout () [0x00001] in D:\agent\1\s\Xamarin.Forms.Core\Page.cs:245 
09-13 12:37:28.203 I/MonoDroid(21323):   at Xamarin.Forms.Page.set_ContainerArea (Xamarin.Forms.Rectangle value) [0x00021] in D:\agent\1\s\Xamarin.Forms.Core\Page.cs:142 
09-13 12:37:28.203 I/MonoDroid(21323):   at Xamarin.Forms.Platform.Android.AppCompat.NavigationPageRenderer.OnLayout (System.Boolean changed, System.Int32 l, System.Int32 t, System.Int32 r, System.Int32 b) [0x000b1] in D:\agent\1\s\Xamarin.Forms.Platform.Android\AppCompat\NavigationPageRenderer.cs:421 
09-13 12:37:28.203 I/MonoDroid(21323):   at Xamarin.Forms.Platform.Android.FormsViewGroup.n_OnLayout_ZIIII (System.IntPtr jnienv, System.IntPtr native__this, System.Boolean p0, System.Int32 p1, System.Int32 p2, System.Int32 p3, System.Int32 p4) [0x0000a] in D:\agent\1\s\Xamarin.Forms.Platform.Android.FormsViewGroup\obj\Debug\generated\src\Xamarin.Forms.Platform.Android.FormsViewGroup.cs:198 
09-13 12:37:28.203 I/MonoDroid(21323):   at (wrapper dynamic-method) Android.Runtime.DynamicMethodNameCounter.33(intptr,intptr,bool,int,int,int,int)
09-13 12:37:28.226 D/Mono    (21323): DllImport searching in: '__Internal' ('(null)').
09-13 12:37:28.226 D/Mono    (21323): Searching for 'java_interop_jnienv_throw'.
09-13 12:37:28.226 D/Mono    (21323): Probing 'java_interop_jnienv_throw'.
09-13 12:37:28.226 D/Mono    (21323): Found as 'java_interop_jnienv_throw'.
**System.ObjectDisposedException:** 'Cannot access a disposed object.
Object name: 'Xamarin.Forms.Platform.Android.Platform+DefaultRenderer'.'

09-13 12:37:30.755 I/MonoDroid(21323): UNHANDLED EXCEPTION:
09-13 12:37:30.758 I/MonoDroid(21323): System.ObjectDisposedException: Cannot access a disposed object.
09-13 12:37:30.759 I/MonoDroid(21323): Object name: 'Xamarin.Forms.Platform.Android.Platform+DefaultRenderer'.
09-13 12:37:30.759 I/MonoDroid(21323):   at (wrapper dynamic-method) Android.Runtime.DynamicMethodNameCounter.33(intptr,intptr,bool,int,int,int,int)
09-13 12:37:30.759 I/MonoDroid(21323):   at (wrapper native-to-managed) Android.Runtime.DynamicMethodNameCounter.33(intptr,intptr,int,int,int,int,int)
09-13 12:37:30.759 I/MonoDroid(21323):   at (wrapper managed-to-native) Java.Interop.NativeMethods.java_interop_jnienv_call_nonvirtual_void_method_a(intptr,intptr&,intptr,intptr,intptr,intptr)
09-13 12:37:30.759 I/MonoDroid(21323):   at Java.Interop.JniEnvironment+InstanceMethods.CallNonvirtualVoidMethod (Java.Interop.JniObjectReference instance, Java.Interop.JniObjectReference type, Java.Interop.JniMethodInfo method, Java.Interop.JniArgumentValue* args) [0x00079] in <42748fcc36b74733af2d9940a8f3cc8e>:0 
09-13 12:37:30.759 I/MonoDroid(21323):   at Java.Interop.JniPeerMembers+JniInstanceMethods.InvokeVirtualVoidMethod (System.String encodedMember, Java.Interop.IJavaPeerable self, Java.Interop.JniArgumentValue* parameters) [0x0005d] in <42748fcc36b74733af2d9940a8f3cc8e>:0 
09-13 12:37:30.759 I/MonoDroid(21323):   at Xamarin.Forms.Platform.Android.FormsViewGroup.MeasureAndLayout (System.Int32 p0, System.Int32 p1, System.Int32 p2, System.Int32 p3, System.Int32 p4, System.Int32 p5) [0x00088] in D:\agent\1\s\Xamarin.Forms.Platform.Android.FormsViewGroup\obj\Debug\generated\src\Xamarin.Forms.Platform.Android.FormsViewGroup.cs:181 
09-13 12:37:30.759 I/MonoDroid(21323):   at Xamarin.Forms.Platform.Android.VisualElementTracker.UpdateLayout () [0x0015c] in D:\agent\1\s\Xamarin.Forms.Platform.Android\VisualElementTracker.cs:111 
09-13 12:37:30.759 I/MonoDroid(21323):   at Xamarin.Forms.Platform.Android.VisualElementRenderer`1[TElement].UpdateLayout () [0x00014] in D:\agent\1\s\Xamarin.Forms.Platform.Android\VisualElementRenderer.cs:139 
09-13 12:37:30.759 I/MonoDroid(21323):   at Xamarin.Forms.Platform.Android.AppCompat.Platform.Xamarin.Forms.Platform.Android.IPlatformLayout.OnLayout (System.Boolean changed, System.Int32 l, System.Int32 t, System.Int32 r, System.Int32 b) [0x0002f] in D:\agent\1\s\Xamarin.Forms.Platform.Android\AppCompat\Platform.cs:235 
09-13 12:37:30.759 I/MonoDroid(21323):   at Xamarin.Forms.Platform.Android.PlatformRenderer.OnLayout (System.Boolean changed, System.Int32 l, System.Int32 t, System.Int32 r, System.Int32 b) [0x0001d] in D:\agent\1\s\Xamarin.Forms.Platform.Android\PlatformRenderer.cs:75 
09-13 12:37:30.759 I/MonoDroid(21323):   at Android.Views.ViewGroup.n_OnLayout_ZIIII (System.IntPtr jnienv, System.IntPtr native__this, System.Boolean changed, System.Int32 l, System.Int32 t, System.Int32 r, System.Int32 b) [0x00008] in <7d2292394f8c488b97f5bc2a0ac0240d>:0 
09-13 12:37:30.759 I/MonoDroid(21323):   at (wrapper dynamic-method) Android.Runtime.DynamicMethodNameCounter.27(intptr,intptr,bool,int,int,int,int)
**System.ObjectDisposedException:** 'Cannot access a disposed object.
Object name: 'Xamarin.Forms.Platform.Android.Platform+DefaultRenderer'.'

09-13 12:37:33.822 E/mono    (21323): 
09-13 12:37:33.822 E/mono    (21323): Unhandled Exception:
09-13 12:37:33.822 E/mono    (21323): System.ObjectDisposedException: Cannot access a disposed object.
09-13 12:37:33.822 E/mono    (21323): Object name: 'Xamarin.Forms.Platform.Android.Platform+DefaultRenderer'.
09-13 12:37:33.822 E/mono    (21323):   at (wrapper dynamic-method) Android.Runtime.DynamicMethodNameCounter.33(intptr,intptr,bool,int,int,int,int)
09-13 12:37:33.822 E/mono    (21323):   at (wrapper native-to-managed) Android.Runtime.DynamicMethodNameCounter.33(intptr,intptr,int,int,int,int,int)
09-13 12:37:33.822 E/mono    (21323):   at (wrapper managed-to-native) Java.Interop.NativeMethods.java_interop_jnienv_call_nonvirtual_void_method_a(intptr,intptr&,intptr,intptr,intptr,intptr)
09-13 12:37:33.822 E/mono    (21323):   at Java.Interop.JniEnvironment+InstanceMethods.CallNonvirtualVoidMethod (Java.Interop.JniObjectReference instance, Java.Interop.JniObjectReference type, Java.Interop.JniMethodInfo method, Java.Interop.JniArgumentValue* args) [0x00079] in <42748fcc36b74733af2d9940a8f3cc8e>:0 
09-13 12:37:33.822 E/mono    (21323):   at Java.Interop.JniPeerMembers+JniInstanceMethods.InvokeVirtualVoidMethod (System.String encodedMember, Java.Interop.IJavaPeerable self, Java.Interop.JniArgumentValue* parameters) [0x0005d] in <42748fcc36b74733af2d9940a8f3cc8e>:0 
09-13 12:37:33.822 E/mono    (21323):   at Xamarin.Forms.Platform.Android.FormsViewGroup.MeasureAndLayout (System.Int32 p0, System.Int32 p1, System.Int32 p2, System.Int32 p3, System.Int32 p4, System.Int32 p5) [0x00088] in D:\agent\1\s\Xamarin.Forms.Platform.Android.FormsViewGroup\obj\Debug\generated\src\Xamarin.Forms.Platform.Android.FormsViewGroup.cs:181 
09-13 12:37:33.822 E/mono    (21323):   at Xamarin.Forms.Platform.Android.VisualElementTracker.UpdateLayout () [0x0015c] in D:\agent\1\s\Xamarin.Forms.Platform.Android\VisualElementTracker.cs:111 
09-13 12:37:33.822 E/mono    (21323):   at Xamarin.Forms.Platform.Android.VisualElementRenderer`1[TElement].UpdateLayout () [0x00014] in D:\agent\1\s\Xamarin.Forms.Platform.Android\VisualElementRenderer.cs:139 
09-13 12:37:33.822 E/mono    (21323):   at Xamarin.Forms.Platform.Android.AppCompat.Platform.Xamarin.Forms.Platform.Android.IPlatformLayout.OnLayout (System.Boolean changed, System.Int32 l, System.Int32 t, System.Int32 r, System.Int32 b) [0x0002f] in D:\agent\1\s\Xamarin.Forms.Platform.Android\AppCompat\Platform.cs:235 
09-13 12:37:33.822 E/mono    (21323):   at Xamarin.Forms.Platform.Android.PlatformRenderer.OnLayout (System.Boolean changed, System.Int32 l, System.Int32 t, System.Int32 r, System.Int32 b) [0x0001d] in D:\agent\1\s\Xamarin.Forms.Platform.Android\PlatformRenderer.cs:75 
09-13 12:37:33.822 E/mono    (21323):   at Android.Views.ViewGroup.n_OnLayout_ZIIII (System.IntPtr jnienv, System.IntPtr native__this, System.Boolean changed, System.Int32 l, System.Int32 t, System.Int32 r, System.Int32 b) [0x00008] in <7d2292394f8c488b97f5bc2a0ac0240d>:0 
09-13 12:37:33.822 E/mono    (21323):   at (wrapper dynamic-method) Android.Runtime.DynamicMethodNameCounter.27(intptr,intptr,bool,int,int,int,int)
09-13 12:37:33.822 E/mono    (21323): --- End of stack trace from previous location where exception was thrown ---
09-13 12:37:33.822 E/mono    (21323): 
09-13 12:37:33.822 E/mono    (21323):   at Java.Interop.JniEnvironment+Strings.NewString (System.Char* unicodeChars, System.Int32 length) [0x0001e] in <42748fcc36b74733af2d9940a8f3cc8e>:0 
09-13 12:37:33.822 E/mono    (21323):   at Java.Interop.JniEnvironment+Strings.NewString (System.String value) [0x00024] in <42748fcc36b74733af2d9940a8f3cc8e>:0 
09-13 12:37:33.822 E/mono    (21323):   at Android.Runtime.JNIEnv.NewString (System.String text) [0x00009] in <7d2292394f8c488b97f5bc2a0ac0240d>:0 
09-13 12:37:33.822 E/mono    (21323):   at Java.Lang.Error..ctor (System.String message) [0x0001f] in <7d2292394f8c488b97f5bc2a0ac0240d>:0 
09-13 12:37:33.822 E/mono    (21323):   at Android.Runtime.JavaProxyThrowable..ctor (System.Exception innerException) [0x00007] in <7d2292394f8c488b97f5bc2a0ac0240d>:0 
09-13 12:37:33.822 E/mono    (21323):   at Java.Lang.Throwable.FromException (System.Exception e) [0x0001d] in <7d2292394f8c488b97f5bc2a0ac0240d>:0 
09-13 12:37:33.822 E/mono    (21323):   at Android.Runtime.AndroidEnvironment.UnhandledException (System.Exception e) [0x00067] in <7d2292394f8c488b97f5bc2a0ac0240d>:0 
09-13 12:37:33.822 E/mono    (21323):   at (wrapper dynamic-method) Android.Runtime.DynamicMethodNameCounter.27(intptr,intptr,bool,int,int,int,int)
09-13 12:37:33.822 E/mono    (21323):   at (wrapper native-to-managed) Android.Runtime.DynamicMethodNameCounter.27(intptr,intptr,int,int,int,int,int)
09-13 12:37:33.826 E/mono-rt (21323): [ERROR] FATAL UNHANDLED EXCEPTION: System.ObjectDisposedException: Cannot access a disposed object.
09-13 12:37:33.826 E/mono-rt (21323): Object name: 'Xamarin.Forms.Platform.Android.Platform+DefaultRenderer'.
09-13 12:37:33.826 E/mono-rt (21323):   at (wrapper dynamic-method) Android.Runtime.DynamicMethodNameCounter.33(intptr,intptr,bool,int,int,int,int)
09-13 12:37:33.826 E/mono-rt (21323):   at (wrapper native-to-managed) Android.Runtime.DynamicMethodNameCounter.33(intptr,intptr,int,int,int,int,int)
09-13 12:37:33.826 E/mono-rt (21323):   at (wrapper managed-to-native) Java.Interop.NativeMethods.java_interop_jnienv_call_nonvirtual_void_method_a(intptr,intptr&,intptr,intptr,intptr,intptr)
09-13 12:37:33.826 E/mono-rt (21323):   at Java.Interop.JniEnvironment+InstanceMethods.CallNonvirtualVoidMethod (Java.Interop.JniObjectReference instance, Java.Interop.JniObjectReference type, Java.Interop.JniMethodInfo method, Java.Interop.JniArgumentValue* args) [0x00079] in <42748fcc36b74733af2d9940a8f3cc8e>:0 
09-13 12:37:33.826 E/mono-rt (21323):   at Java.Interop.JniPeerMembers+JniInstanceMethods.InvokeVirtualVoidMethod (System.String encodedMember, Java.Interop.IJavaPeerable self, Java.Interop.JniArgumentValue* parameters) [0x0005d] in <42748fcc36b74733af2d9940a8f3cc8e>:0 
09-13 12:37:33.826 E/mono-rt (21323):   at Xamarin.Forms.Platform.Android.FormsViewGroup.MeasureAndLayout (System.Int32 p0, System.Int32 p1, System.Int32 p2, System.Int32 p3, System.Int32 p4, System.Int32 p5) [0x00088] in D:\agent\1\s\Xamarin.Forms.Platform.Android.FormsViewGroup\obj\Debug\generated\src\Xamarin.Forms.Platform.Android.FormsViewGroup.cs:181 
09-13 12:37:33.826 E/mono-rt (21323):   at Xamarin.Forms.Platform.Android.VisualElementTracker.UpdateLayout () [0x0015c] in D:\agent\1\s\Xamarin.Forms.Platform.Android\VisualElementTracker.cs:111 
09-13 12:37:33.826 E/mono-rt (21323):   at Xamarin.Forms.Platform.Android.VisualElementRenderer`1[TElement].UpdateLayout () [0x00014] in D:\agent\1\s\Xamarin.Forms.Platform.Android\VisualElementRenderer.cs:139 
09-13 12:37:33.826 E/mono-rt (21323):   at Xamarin.Forms.Platform.Android.AppCompat.Platform.Xamarin.Forms.Platform.Android.IPlatformLayout.OnLayout (System.Boolean changed, System.Int32 l, System.Int32 t, System.Int32 r, System.Int32 b) [0x0002f] in D:\agent\1\s\Xamarin.Forms.Platform.Android\AppCompat\Platform.cs:235 
09-13 12:37:33.826 E/mono-rt (21323):   at Xamarin.Forms.Platform.Android.PlatformRenderer.OnLayout (System.Boolean changed, System.Int32 l, System.Int32 t, System.Int32 r, System.Int32 b) [0x0001d] in D:\agent\1\s\Xamarin.Forms.Platform.Android\PlatformRenderer.cs:75 
09-13 12:37:33.826 E/mono-rt (21323):   at Android.Views.ViewGroup.n_OnLayout_ZIIII (System.IntPtr jnienv, System.IntPtr native__this, System.Boolean changed, System.Int32 l, System.Int32 t, System.Int32 r, System.Int32 b) [0x00008] in <7d2292394f8c488b97f5bc2a0ac0240d>:0 
09-13 12:37:33.826 E/mono-rt (21323):   at (wrapper dynamic-method) Android.Runtime.DynamicMethodNameCounter.27(intptr,intptr,bool,int,int,int,int)
09-13 12:37:33.826 E/mono-rt (21323): --- End of stack trace from previous location where exception was thrown ---
09-13 12:37:33.826 E/mono-rt (21323): 
09-13 12:37:33.826 E/mono-rt (21323):   at Java.Interop.JniEnvironment+Strings.NewString (System.Char* unicodeChars, System.Int32 length) [0x0001e] in <42748fcc36b74733af2d9940a8f3cc8e>:0 
09-13 12:37:33.826 E/mono-rt (21323):   at Java.Interop.JniEnvironment+Strings.NewString (System.String value) [0x00024] in <42748fcc36b74733af2d9940a8f3cc8e>:0 
09-13 12:37:33.826 E/mono-rt (21323):   at Android.Runtime.JNIEnv.NewString (System.String text) [0x00009] in <7d2292394f8c488b97f5bc2a0ac0240d>:0 
09-13 12:37:33.826 E/mono-rt (21323):   at Java.Lang.Error..ctor (System.String message) [0x0001f] in <7d2292394f8c488b97f5bc2a0ac0240d>:0 
09-13 12:37:33.826 E/mono-rt (21323):   at Android.Runtime.JavaProxyThrowable..ctor (System.Exception innerException) [0x00007] in <7d2292394f8c488b97f5bc2a0ac0240d>:0 
09-13 12:37:33.826 E/mono-rt (21323):   at Java.Lang.Throwable.FromException (System.Exception e) [0x0001d] in <7d2292394f8c488b97f5bc2a0ac0240d>:0 
09-13 12:37:33.826 E/mono-rt (21323):   at Android.Runtime.AndroidEnvironment.UnhandledException (System.Exception e) [0x00067] in <7d2292394f8c488b97f5bc2a0ac0240d>:0 
09-13 12:37:33.826 E/mono-rt (21323):   at (wrapper dynamic-method) Android.Runtime.DynamicMethodNameCounter.27(intptr,intptr,bool,int,int,int,int)terminating

09-13 12:37:33.826 E/mono-rt (21323):   at (wrapper native-to-managed) Android.Runtime.DynamicMethodNameCounter.27(intptr,intptr,int,int,int,int,int)
09-13 12:37:33.865 E/libc++abi(21323): terminating
09-13 12:37:33.866 F/libc    (21323): Fatal signal 6 (SIGABRT), code -1 (SI_QUEUE) in tid 21323 (ame.labelrender), pid 21323 (ame.labelrender)

--- a/Xamarin.Forms.Platform.Android/Platform.cs
+++ b/Xamarin.Forms.Platform.Android/Platform.cs
@@ -1335,8 +1335,12 @@ namespace Xamarin.Forms.Platform.Android
 		{
 			Performance.Start(out string reference);
 
-			// FIXME: potential crash
 			IVisualElementRenderer visualElementRenderer = GetRenderer(view);
+
+			if (visualElementRenderer == null || visualElementRenderer.View.IsDisposed())
+			{
+				return new SizeRequest(Size.Zero, Size.Zero);
+			}
 
 			var context = visualElementRenderer.View.Context;
 

